### PR TITLE
Rails 5

### PR DESCRIPTION
**Pull request by lucaspiller**
Support for Rails 5. WIP.

Outstanding issues:
- [x] DEPRECATION WARNING: #table_exists? currently checks both tables and views. This behavior is deprecated and will be changed with Rails 5.1 to only check tables. Use #data_source_exists? instead. (called from chrono_metadata_for at lib/chrono_model/adapter.rb:633)
- [x] DEPRECATION WARNING: Passing a column to `quote` has been deprecated. It is only used for type casting, which should be handled elsewhere. See https://github.com/rails/arel/commit/6160bfbda1d1781c3b08a33ec4955f170e95be11 for more information. (called from time_for_time_query at lib/chrono_model/time_machine.rb:395)
- [x] Setup Travis to run tests against Rails 4 and 5
- [x] Fix specs

### CODE REVIEW

File: lib/active_record/connection_adapters/chronomodel_adapter.rb [hunk 1]
```diff
@@ -17,7 +17,12 @@ def chronomodel_connection(config) # :nodoc:
       conn_params[:dbname] = conn_params.delete(:database) if conn_params[:database]
 
       # Forward only valid config params to PGconn.connect.
-      conn_params.keep_if { |k, _| VALID_CONN_PARAMS.include?(k) }
+      valid_conn_param_keys = if ActiveRecord::VERSION::MAJOR == 4
+                                VALID_CONN_PARAMS
+                              else
+                                PGconn.conndefaults_hash.keys + [:requiressl]
```
Comment (vjt) [inline lib/active_record/connection_adapters/chronomodel_adapter.rb]: Nice

**Comment by lucaspiller** [comment]
@vjt Can you take a look at the deprecation of `quote`, I'm not exactly sure how it's being used and what it can be safely changed to. I assume it'll need to be a conditional like the other bits, so it's probably worth extracting it out to a `chrono_quote` method.

**Comment by vjt** [comment]
@lucaspiller it is inconsistently used in place of `quote_value` and `quote_table`. https://github.com/ifad/chronomodel/search?utf8=%E2%9C%93&q=quote

I do not think we need a wrapper, just to use the correct methods for the context.

**Comment by kirillsalykin** [comment]
Hi @lucaspiller @vjt ,

I plan to use chronomodel with Rails 5, thus I need this PR. I'd like to help to finish it, could you please point me what I can pick? Thanks.

**Comment by vjt** [comment]
@kirillsalykin thanks for proposing to help.

Chronomodel functions, except for STI, are already working with Rails5, but for some deprecation warnings. A full report is available [on Travis](https://travis-ci.org/ifad/chronomodel/jobs/203717585).

* The [deprecation warning on `quote`](https://travis-ci.org/ifad/chronomodel/jobs/203717585#L308) is located [in this code](https://github.com/ifad/chronomodel/blob/rails5/lib/chrono_model/adapter.rb#L704) that is using [ActiveRecord::ConnectionAdapters::Quoting#quote](https://github.com/rails/rails/blob/d1200a8911f700fa8a54b751286f9437b94340fb/activerecord/lib/active_record/connection_adapters/abstract/quoting.rb#L9). It should be understood whether `column.default` is already quoted in Rails 5, or whether using [`quote_string`](https://github.com/rails/rails/blob/d1200a8911f700fa8a54b751286f9437b94340fb/activerecord/lib/active_record/connection_adapters/abstract/quoting.rb#L80) is enough in this context.
* Failures from 1 to 17 are all related to `select_value` now returning a Ruby native TrueClass or FalseClass type instead of the underlying database-specific `'t'` or `'f'`. The [table matchers](https://github.com/ifad/chronomodel/blob/rails5/spec/support/matchers/table.rb) should be updated to check equality against `true` and `false` instead of `'t'` and `'f'`, and as well the [json_ops_spec.rb](https://github.com/ifad/chronomodel/blob/rails5/spec/json_ops_spec.rb#L17).
* The _indexes_ part of failures [2](https://github.com/ifad/chronomodel/blob/rails5/spec/json_ops_spec.rb#L17), [6](https://travis-ci.org/ifad/chronomodel/jobs/203717585#L350) and [11](https://travis-ci.org/ifad/chronomodel/jobs/203717585#L394) are instead related to the behaviour of [`select_values` as used in this matcher](https://github.com/ifad/chronomodel/blob/rails5/spec/support/matchers/table.rb#L141), it needs to be seen how its output has changed, and then update the test.
* Failure [18](https://travis-ci.org/ifad/chronomodel/jobs/203717585#L539) is the toughest. This requires analysing AR code in how it now deals with STI. I have no clue at the moment, this requires a debug session.

Please let me know if I can be of further help.

File: .travis.yml [hunk 2]
```diff
@@ -18,4 +26,5 @@ before_script:
   - psql -c "CREATE DATABASE chronomodel;" -U postgres
 
 script:
-  - bundle exec rake TEST_CONFIG=./spec/config.travis.yml CODECLIMATE_REPO_TOKEN=dedfb7472ee410eec459bff3681d9a8fd8dd237e9bd7e8675a7c8eb7e253bba9
+  - bundle exec rake TEST_CONFIG=./spec/config.travis.yml
```
Comment (vjt) [inline .travis.yml]: This should be amended as per https://docs.codeclimate.com/v1.0/docs/travis-ci-ruby-test-coverage

**Pull request merged**